### PR TITLE
Add support for Protomap tiles

### PR DESCRIPTION
**Pull request by halocline**

### CODE REVIEW

File: grommet-leaflet/src/layers/Map/Map.jsx [hunk 1]
```diff
@@ -22,6 +23,7 @@ const Map = forwardRef(
       scrollWheelZoom = true,
       tileLayer,
       theme,
+      maxZoom = 18,
```
Comment (taysea) [inline grommet-leaflet/src/layers/Map/Map.jsx:26]: Can you add a comment for what "18" equates to? Or a link to the legen

File: grommet-leaflet/src/layers/Map/Map.jsx [hunk 2]
```diff
@@ -40,12 +42,18 @@ const Map = forwardRef(
           center={center}
           ref={ref}
           scrollWheelZoom={scrollWheelZoom}
+          maxZoom={maxZoom}
```
Comment (taysea) [inline grommet-leaflet/src/layers/Map/Map.jsx:45]: Alphabetize

File: grommet-leaflet/src/themes/base.js [hunk 3]
```diff
@@ -79,6 +79,41 @@ const base = {
     pad: 'small',
     round: 'xsmall',
   },
+  // tiles: {
+  // Protomaps [basemap layers](https://docs.protomaps.com/basemaps/layers#basemap-layers) derived from OpenStreetMap.
```
Comment (taysea) [inline grommet-leaflet/src/themes/base.js:83]: General question, how were these colors selected?

Also, should we be considering what a light/dark theme structure would look like before we commit to this theme structure?

File: grommet-leaflet/src/layers/Map/Map.jsx [hunk 4]
```diff
@@ -57,9 +65,11 @@ Map.propTypes = {
   center: PropTypes.array,
   children: PropTypes.any,
   scrollWheelZoom: PropTypes.string || PropTypes.bool,
+  maxZoom: PropTypes.number,
   tileLayer: PropTypes.shape({
     attribution: PropTypes.string,
     url: PropTypes.string.isRequired,
+    format: PropTypes.oneOf(['pmtiles']),
```
Comment (taysea) [inline grommet-leaflet/src/layers/Map/Map.jsx:72]: Attribution is marked as required in typescript definition, but not here. Just need to align them.

File: grommet-leaflet/src/layers/TileLayer/TileLayer.jsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      />
+    );
+  }
+
+  return tileLayer;
+};
+
+TileLayer.propTypes = {
+  attribution: PropTypes.string,
+  url: PropTypes.string,
```
Comment (taysea) [inline grommet-leaflet/src/layers/TileLayer/TileLayer.jsx:48]: Is this correct?
```suggestion
  url: PropTypes.string.isRequired,
```

File: demo-app/src/pages/default/index.jsx [hunk 6]
```diff
@@ -9,6 +9,18 @@ import {
   ReverseAnchor,
 } from '../../components';
 
+const tileLayer = {
+  // url: 'https://tiles.stadiamaps.com/tiles/alidade_smooth/{z}/{x}/{y}{r}.png',
```
Comment (taysea) [inline demo-app/src/pages/default/index.jsx:13]: Remove comment

File: demo-app/src/pages/default/index.jsx [hunk 7]
```diff
@@ -9,6 +9,18 @@ import {
...
+  // attribution: `
+  //   &copy; <a href="https://stadiamaps.com/">Stadia Maps</a>,
+  //   &copy; <a href="https://openmaptiles.org/">OpenMapTiles</a>
+  //   &copy; <a href="http://openstreetmap.org">OpenStreetMap</a> contributors`,
+  format: 'pmtiles',
+  url: 'http://localhost:8080/planet/{z}/{x}/{y}.mvt',
```
Comment (taysea) [inline demo-app/src/pages/default/index.jsx:19]: Does someone testing the demo-app have to do anything special to get this server running locally? or does it automatically start with the `dev` script?

Just wondering from a dev experience of someone looking at the project.

File: grommet-leaflet/src/layers/Map/Map.jsx (hunk 1 above)
Comment (halocline) [reply grommet-leaflet/src/layers/Map/Map.jsx:26]: added comments and link to docs

File: grommet-leaflet/src/layers/Map/Map.jsx (hunk 2 above)
Comment (halocline) [reply grommet-leaflet/src/layers/Map/Map.jsx:45]: done.

File: grommet-leaflet/src/themes/base.js (hunk 3 above)
Comment (halocline) [reply grommet-leaflet/src/themes/base.js:83]: Added TODO comment

File: demo-app/src/pages/default/index.jsx (hunk 6 above)
Comment (halocline) [reply demo-app/src/pages/default/index.jsx:13]: modified comments to communicate intent (e.g. PMTiles won't work until we get a PMTiles file on CDN)

File: demo-app/src/pages/default/index.jsx (hunk 7 above)
Comment (halocline) [reply demo-app/src/pages/default/index.jsx:19]: Modified comment to address as a temp fix

**Pull request merged**